lutris/util/graphics/drivers.py

```diff
@@ -2,83 +2,139 @@
+            "arch": arch,
+            "version": glx_info.opengl_version.rsplit(maxsplit=1)[-1],  # type: ignore[attr-defined]
+        }
+    }
 
 
-def get_nvidia_gpu_ids():
+def get_nvidia_gpu_ids() -> List[str]:
     """Return the list of Nvidia GPUs"""
-    return os.listdir("/proc/driver/nvidia/gpus")
+    with contextlib.suppress(OSError):
+        return os.listdir("/proc/driver/nvidia/gpus")
+    values = read_process_output(
+        # 10de is NVIDIA's vendor ID, 0300 gets you video controllers.
+        ["lspci", "-D", "-n", "-d", "10de::0300"],
+    ).splitlines()
+    return [line.split(maxsplit=1)[0] for line in values]
 
 
-def get_nvidia_gpu_info(gpu_id):
+def get_nvidia_gpu_info(gpu_id: str) -> Dict[str, str]:
     """Return details about a GPU"""
-    with open("/proc/driver/nvidia/gpus/%s/information" % gpu_id, encoding='utf-8') as info_file:
-        content = info_file.readlines()
-    infos = {}
-    for line in content:
+    with contextlib.suppress(OSError):
+        with open(
+            "/proc/driver/nvidia/gpus/%s/information" % gpu_id, encoding="utf-8"
+        ) as info_file:
+            content = info_file.readlines()
+        infos = {}
+        for line in content:
+            key, value = line.split(":", 1)
+            infos[key] = value.strip()
+        return infos
+    lspci_data = read_process_output(["lspci", "-v", "-s", gpu_id])
+    model_info = re.search(r"NVIDIA Corporation \w+ \[(.+?)\]", lspci_data)
+    if model_info:
+        model = model_info.group(1)
+    irq_info = re.search("IRQ ([0-9]+)", lspci_data)
+    if irq_info:
+        irq = irq_info.group(1)
+
+    info = {
+        "Model": f"NVIDIA {model}",
+        "IRQ": irq,
+        "Bus Location": gpu_id,
+    }
+    for line in lspci_data.splitlines():
+        if ":" not in line:
+            continue
         key, value = line.split(":", 1)
-        infos[key] = value.strip()
-    return infos
+        info[key.strip()] = value.strip()
+    return info
 
 
-def is_nvidia():
+def is_nvidia() -> bool:
     """Return true if the Nvidia drivers are currently in use"""
-    return os.path.exists("/proc/driver/nvidia")
+    with contextlib.suppress(OSError):
+        return os.path.exists("/proc/driver/nvidia")
+    with contextlib.suppress(OSError):
+        with open("/proc/modules") as f:
+            modules = f.read()
+        return bool(re.search(r"^nvidia ", modules, flags=re.MULTILINE))
+    glx_info = GlxInfo()
+    return "NVIDIA" in glx_info.opengl_vendor  # type: ignore[attr-defined]
 
 
-def get_gpus():
+def get_gpus() -> Iterable[str]:
     """Return GPUs connected to the system"""
     if not os.path.exists("/sys/class/drm"):
         logger.error("No GPU available on this system!")
-        return
+        return []
     try:
         cardlist = os.listdir("/sys/class/drm/")
     except PermissionError:
-        logger.error("Your system does not allow reading from /sys/class/drm, no GPU detected.")
```

Comment:
I like this better, but you've been catching OSError.

What's the story here? Your PR comment makes me thing you really want PermissionError, but perhaps it's raising just OSError? Maybe we ought to change all these except blocks to catch OSError for safety?

Reply:
This is probably a case where catching OSError might be better (e.g. what happens if `/sys/class/drm` is a file for some reason? We might not want to let the exception bubble up all the way to the top, but it's still an error detecting the GPU), but I'll defer to you (and change it if you agree).